src/App.vue

```diff
@@ -2,19 +2,35 @@
   <div id="app" :class="$style.page">
     <the-header />
     <main :class="$style.main">
-      <router-view />
+      <router-animation :direction="direction">
+        <router-view />
+      </router-animation>
     </main>
   </div>
 </template>
 
 <script>
 import TheHeader from '@/components/TheHeader.vue'
+import RouterAnimation from '@/components/Animations/RouterAnimation.vue'
 
 export default {
   name: 'App',
   components: {
+    RouterAnimation,
     TheHeader,
   },
+  data() {
+    return {
+      direction: '',
+    }
+  },
+  watch: {
+    // watch the `$route` to determine the transition to use
+    $route(to) {
+      const isListPage = to.name === 'Root' || to.name === 'City'
+      this.direction = isListPage ? 'left' : 'right'
```

Comment:
Nit: The following sequence is a bit weird:

1. Start at the homepage
2. Click on incident
3. Click back to list
4. Click browser back button

The transition from 3->4 is a right animation, but it's a bit jarring because I'm clicking the browser back button which is pointing to the left. Probably the forward button could have a similar problem.

This is totally nitpicky feedback though so feel free to ignore it.

Reply:
I dont know how to fix this. But because of these kind of issues, I'm not sure this update really is an improvement.

Reply:
It might be nice to disable animation when going forward or back with the browser button

Reply:
Great addition. I'll dive into that.

Reply:
> [Watch out for this](https://developer.mozilla.org/en-US/docs/Web/API/Window/popstate_event#The_history_stack:~:text=Browsers%20tend%20to%20handle%20the%20popstate,on%20page%20load%2C%20but%20Firefox%20doesn't.)
> 
> > Browsers tend to handle the popstate event differently on page load. Chrome (prior to v34) and Safari always emit a popstate event on page load, but Firefox doesn't.

It works in Chrome and Firefox, but I'm still having weird issues in Safari. Hope I can fix it.